mirrord/layer/src/tcp_steal.rs

```diff
@@ -230,8 +228,20 @@ impl TcpHandler for TcpStealHandler {
             return Ok(());
         }
 
-        let steal_type = if self.http_ports.contains(&original_port) && let Some(filter_str) = self.http_filter.take() {
-            FilteredHttp(request_port, Filter::new(filter_str)?)
+        let steal_type = if self.http_filter_settings.ports.contains(&original_port) {
+            match self.http_filter_settings.filter {
+                LayerHttpFilter::None => All(request_port),
```

Comment:
Thought about existing code, not this PR: It's a bit weird our config allows http filter ports to be specified without a filter. Maybe we can add a todo to change the config such that it's not possible on the type system level, so that we don't have to handle those useless situations.

Reply:
not sure where I should add this todo?